### PR TITLE
Transaction type files

### PR DESCRIPTION
**Pull request by jimni1222**
## Proposed changes

This PR introduces transaction type files.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-js/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-js)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: types/packages/caver-transaction/src/transactionTypes/abstractFeeDelegatedTransaction.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import SignatureData from '../../../caver-wallet/src/keyring/signatureData'
+import AbstractTransaction, { CreateTransactionObject } from './abstractTransaction'
+
+export default class AbstractFeeDelegatedTransaction extends AbstractTransaction {
+    constructor(typeString: string, createTxObj: CreateTransactionObject)
+
+    signAsFeePayer(key: string | Keyring, index?: number, hasher?: Function): Promise<AbstractFeeDelegatedTransaction>
+    appendFeePayerSignatures(signatures: string[] | string[][] | SignatureData | SignatureData[]): void
+    combineSignedRawTransactions(rlpEncodedTxs: string[]): string
+    getSenderTxHash(): string
```
Comment (sirano11) [inline types/packages/caver-transaction/src/transactionTypes/abstractFeeDelegatedTransaction.d.ts:26]: Do we need to define this function? It already define AbstractTransaction.

Comment (jimni1222) [reply types/packages/caver-transaction/src/transactionTypes/abstractFeeDelegatedTransaction.d.ts]: I removed this function!

File: types/packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    getCommonRLPEncodingForSignature(): string
+    recoverPublicKeys(): string[]
+
+    private _to: string
+    public to: string
+    private _value: string
+    public value: string
+    private _input: string
+    public input: string
+    private _data: string
```
Comment (sirano11) [inline types/packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.d.ts:36]: I thought this field didn't define in LeagcyTransaction.

Comment (jimni1222) [reply types/packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.d.ts:36]: In deprecated LegacyTrasnaction type and also Contract uses `data` filed, so in data setter, set value to the `input` in this case.

File: types/packages/caver-transaction/src/transactionTypes/smartContractDeploy/feeDelegatedSmartContractDeploy.d.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    getRLPEncoding(): string
+    getCommonRLPEncodingForSignature(): string
+
+    private _to: string
+    public to: string
+    private _value: string
+    public value: string
+    private _input: string
+    public input: string
+    private _data: string
```
Comment (sirano11) [inline types/packages/caver-transaction/src/transactionTypes/smartContractDeploy/feeDelegatedSmartContractDeploy.d.ts:34]: Also, This field didn't define in FeeDelegatedSmartContractDeploy.

**Review by sirano11 — APPROVED** [review]

**Pull request merged**